### DISCUSSION TITLE
Asking different package ID

### DISCUSSION THREAD

**Discussion by k4waiiiii** [microsoft/winget-cli#3232] [Q&A] [answered]
What is the difference Microsoft.package name with Microsoft.package name_random number

![image](https://user-images.githubusercontent.com/49643347/236832603-782f8442-05b5-4fa0-a7fa-8c4c43127ad6.png)

**Comment by Trenly** [comment, marked as answer]
That is the how the package has registered itself with Windows, and is the Product Family Name for msixbundle packages, or the Product Code for exe/msi based installers. The PFN or Product Codes are shown when winget can't find a match to an existing package in any of the sources. This means that the packages like `Microsoft.WindowsStore_8wekyb3d8bbwe` can't be installed using winget. However, the identifier (PFN in this case) can still be used to uninstall them

**Reply by florelis** [reply]
For a bit more context, the names that include the "random number" are all packaged apps (appx or msix). The string at the end is a hash computed from the publisher name, and I think it's used to ensure you can have multiple packages with the same name but different publishers. `8wekyb3d8bbwe` is obtained from the Microsoft publisher `CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US`
